corehq/apps/app_manager/models.py

```diff
@@ -5307,7 +5309,7 @@ def create_jadjar_from_build_files(self, save=False):
     def timing_context(self):
         return TimingContext(self.name)
 
-    def validate_app(self):
+    def validate_app(self, previous_version=None):
```

Comment:
Can you add a comment on when `previous_version` should or shouldn't be passed? It's unfortunate that it's really needed for making the build files but needs to be passed to `validate_app` which is only sometimes related to making builds.

Also curious why `previous_version` doesn't get passed to the `create_all_builds` call in download.py: https://github.com/dimagi/commcare-hq/blob/d2a1e34105eb6bc27d66914551f31709500a7147/corehq/apps/app_manager/views/download.py#L486

Reply:
Nevermind, this comment isn't relevant after reading https://github.com/dimagi/commcare-hq/pull/22801/commits/bc9bf34c8b06cffb130514de7d99f06668c278db (which cleans up the code quite a bit).